src/configure.ac

```diff
@@ -902,7 +907,16 @@ AC_CHECK_FUNCS([posix_memalign aligned_alloc memalign __mingw_aligned_malloc _al
 AC_CACHE_SAVE
 
 if test "x$simd" != xno ; then
-  CPU_BEST_FLAGS_MAIN=-DJOHN_$(echo ${SIMD_NAME} | tr .a-z _A-Z)
+  if echo ${SIMD_NAME} | ${GREP} -q "AVX512("; then
+    types=$(echo "${SIMD_NAME}" | ${GREP} -o '(.*)' | tr -d "()" | tr "+" "\n")
+
+    for ext in $types
+    do
+      CPU_BEST_FLAGS_MAIN+="-DJOHN_$(echo AVX512$ext) "
+    done
+  else
+    CPU_BEST_FLAGS_MAIN=-DJOHN_$(echo ${SIMD_NAME} | tr .a-z _A-Z)
+  fi
```

Comment:
I doubt we need this complication. Can't we just continue with `JOHN_AVX512BW` alone, but understand that it implies VL and DQ? I also don't know whether the `+=` syntax works with other shells.

Reply:
This is possible if everyone understands that BW implies the rest.